### PR TITLE
Reshuffle font code around

### PR DESCRIPTION
**Pull request by dpwiz**

### CODE REVIEW

File: src/DearImGui.hs [hunk 1]
```diff
@@ -86,6 +86,11 @@ module DearImGui
   , pushStyleVar
   , popStyleVar
 
+  , withFont
+  , Raw.Font.pushFont
```
Comment (dpwiz) [inline src/DearImGui.hs:90]: IDK if those "unsafe" functions should actually be reexported by the public API.

**Comment by dpwiz** [comment]
Most of the stuff is my mere preferences I hope you'll find sensible :sweat_smile:

File: examples/vulkan/Main.hs [hunk 2]
```diff
@@ -134,12 +134,6 @@ app = do
     ImGui.createContext
     ImGui.destroyContext
 
-  logDebug "Adding fonts"
-  ImGui.clearFontAtlas
-  _default <- ImGui.addFontDefault
-  _custom <- ImGui.addFontFromFileTTF "imgui/misc/fonts/ProggyTiny.ttf" 10
```
Comment (chekoopa) [inline examples/vulkan/Main.hs:140]: Whoops, skipped that one somehow.

File: src/DearImGui.hs (hunk 1 above)
Comment (chekoopa) [reply src/DearImGui.hs:90]: They are similar to `begin*` and `end*` functions, so I don't see much problem.

**Comment by dpwiz** [comment]
Anything else? Let's merge this and I roll out a fresh dearimgui.

File: src/DearImGui/Raw/Font/GlyphRanges.hs [hunk 3]
```diff
@@ -82,6 +89,109 @@ C.context (Cpp.cppCtx <> C.bsCtx <> imguiContext)
...
+  | Korean
+  | Japanese
+  | ChineseFull
+  | ChineseSimplifiedCommon
+  | Cyrillic
+  | Thai
+  | Vietnamese
+  deriving (Eq, Ord, Show, Enum, Bounded)
+
+getBuiltin :: Builtin -> GlyphRanges
```
Comment (chekoopa) [inline src/DearImGui/Raw/Font/GlyphRanges.hs:110]: Would document later.

File: src/DearImGui/FontAtlas.hs [hunk 4]
```diff
@@ -195,229 +229,265 @@ build = Raw.buildFontAtlas
...
-  bracket acquire (FontConfig.destroy) (fn . Just)
-  where
-    acquire = do
-      config <- FontConfig.new
-      liftIO $ setup config
-      return config
-
+withConfig :: (MonadUnliftIO m) => Maybe ConfigSetup -> (Maybe FontConfig -> m a) -> m a
+withConfig mSetup action =
+  case mSetup of
```
Comment (chekoopa) [inline src/DearImGui/FontAtlas.hs:317]: Why `case` instead of patterns tho

**Review by chekoopa — APPROVED** [review]
Nice and insightful!

**Comment by chekoopa** [comment]
Done.

**Pull request merged**